pkg/overlay/server_test.go

```diff
@@ -65,3 +67,142 @@ func TestServer(t *testing.T) {
 		}
 	}
 }
+
+func TestNewNodeFiltering(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 4, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(planet.Shutdown)
+
+	planet.Start(ctx)
+	// we wait a second for all the nodes to complete bootstrapping off the satellite
+	time.Sleep(2 * time.Second)
+
+	satellite := planet.Satellites[0]
+
+	for i, tt := range []struct {
+		name                  string
+		newNodeAuditThreshold int64
+		newNodePercentage     float64
+		requestedNodeAmt      int64
+		expectedResultLength  int
+		reputableNodes        int
+	}{
+		{
+			name:                  "case: fewer than required reputable nodes",
+			requestedNodeAmt:      4,
+			reputableNodes:        3,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: more than required reputable nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        3,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: zero reputable nodes found, only new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        0,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:              "case: fewer than required new nodes",
+			requestedNodeAmt:  2,
+			reputableNodes:    3,
+			newNodePercentage: 0.5,
+			// this gives extra reputable instead
+			expectedResultLength:  3,
```

Comment:
how is `requestedNodeAmt` 2 but `expectedResultLength` is 3?

Reply:
yeah sorry for not clarifying this sooner (it's kind of confusing) but after talking to JT, we decided the functionality should be like:

`result` =   `requestedNodeAmt` + _requestedNodeAmt * newNodePercentage_

where the part in italics is the amount of new nodes included in the result.

Reply:
ohhhhhhh

Reply:
yeah I think this means I should also rename `maxNodes`